services/Donation.js

```diff
@@ -60,6 +61,112 @@ class DonationService {
       return false;
     }
   }
+
+  /**
+   * @description get total donations
+   */
+  static async getTotal(startDate, endDate) {
+    try {
+      const matchQuery = {};
+      if (startDate & endDate) {
+        matchQuery.createdAt = { $gte: startDate, $lte: endDate };
+      }
+
+      const totalDonations = await Donation.aggregate([
+        { $match: matchQuery },
+        { $group: { _id: null, totalAmount: { $sum: "$transaction.amount" } } }
+      ]);
+
+      return {
+        totalAmount:
+          totalDonations?.length === 0 ? 0 : totalDonations[0].totalAmount
+      };
+    } catch (e) {
+      Logger.error("Error getting total donations", JSON.stringify(e));
+      throw new ErrorHandler(e, 500);
+    }
+  }
+
+  /**
+   * @description get total donations and donors
+   */
+  static async getTotalDonorsAndAmount() {
+    try {
+      const donationStats = await Donation.aggregate([
+        {
+          $group: {
+            _id: null,
+            totalAmount: { $sum: "$transaction.amount" },
+            uniqueDonorsCount: { $addToSet: "$donor.email" },
+            donationCount: { $sum: 1 }
+          }
+        }
+      ]);
+      console.log(donationStats);
+      const result = {
+        totalAmount:
+          donationStats.length > 0 ? donationStats[0].totalAmount : 0,
+        uniqueDonorsCount:
+          donationStats.length > 0
+            ? donationStats[0].uniqueDonorsCount.length
+            : 0,
+        donationCount:
+          donationStats.length > 0 ? donationStats[0].donationCount : 0
+      };
+
+      return result;
+    } catch (e) {
+      Logger.error("Error getting total donations", JSON.stringify(e));
+      throw new ErrorHandler(e, 500);
+    }
+  }
+
+  /**
+   * @description get all donations
+   * @param {Id} cursor
+   * @param {Number} limit
+   */
+  static async getDonations(cursor, limit) {
+    try {
+      // const startIndex = (page - 1) * limit;
+
+      // const donations = await Donation.find(
+      //   {},
+      //   { _id: 0, "donor.name": 1, "transaction.amount": 1, createdAt: 1 }
+      // )
+      //   .sort({ createdAt: -1 })
+      //   .skip(startIndex)
+      //   .limit(limit);
+
+      // return donations;
+      let query = {};
+
+      // if cursor is provided, add to query
+      if (cursor) {
+        query._id = { $lt: cursor };
```

Comment:
@Chiemezuo, what does this do? Are we looking for documents whose id is less than `cursor`?

Reply:
Yes, because we are returning the items in reverse chronological order, the first 50 items on the list would ideally be the 50 most-recently added items. And the second 50 items would the the 50 items whose IDs are less than the id of the 50th item from the first 50.